tests/integration_tests/functional/test_balloon.py

```diff
@@ -119,31 +119,54 @@ def copy_util_to_rootfs(rootfs_path, util):
     subprocess.check_call("rmdir tmpfs", shell=True)
 
 
-def _test_rss_memory_lower(test_microvm):
+def _test_rss_memory_lower(test_microvm, use_legacy_api=False):
     """Check inflating the balloon makes guest use less rss memory."""
     # Get the firecracker pid, and open an ssh connection.
     firecracker_pid = test_microvm.jailer_clone_pid
     ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
 
     # Using deflate_on_oom, get the RSS as low as possible
-    response = test_microvm.balloon.patch(amount_mb=200)
-    assert test_microvm.api_session.is_status_no_content(response.status_code)
+    if use_legacy_api:
+        response = test_microvm.balloon.patch(amount_mb=200)
+        assert test_microvm.api_session.is_status_no_content(
+            response.status_code
+        )
+    else:
+        response = test_microvm.balloon.patch(amount_mib=200)
+        assert test_microvm.api_session.is_status_no_content(
+            response.status_code
+        )
 
     # Get initial rss consumption.
     init_rss = get_stable_rss_mem_by_pid(firecracker_pid)
 
     # Get the balloon back to 0.
-    response = test_microvm.balloon.patch(amount_mb=0)
-    assert test_microvm.api_session.is_status_no_content(response.status_code)
+    if use_legacy_api:
```

Comment:
we can move this `if else`'s to the `create_json` method in the framework, to avoid the duplication, by passing the `use_legacy_api` to that method

Reply:
I don't think it matters much as we will remove this in a couple of days. I initially tried to build this component without duplicating but I found the logic became very convoluted and I just went for the scrappy approach.

If you have a clear suggestion that makes this easier and works, I'll update this, but I don't want to spend more time on developing this particular part of the test.

Reply:
I think I already mentioned my suggestion, but I'm fine keeping this as-is, if we'll remove it soon

Reply:
Created a tracking issue for this.